src/board/system76/common/power.c

```diff
@@ -357,14 +386,9 @@ void power_cpu_reset(void) {
     fan_reset();
     // Reset KBC and touchpad states
     kbled_reset();
-    // Set PL4
+    // Set PsysPL2
     //TODO: if this returns false, retry?
-    power_peci_limit(
-        // AC is connected
-        (!gpio_get(&ACIN_N)) &&
-        // There is available current
-        (battery_charger_input_current >= CHARGER_INPUT_CURRENT)
-    );
+    power_peci_limit(gpio_get(&ACIN_N));
```

Comment:
If this is done right after CPU reset, BIOS will not override those values? Or values set by PECI are stored elsewhere and take precedence?

Reply:
The BIOS will override these values. Ideally we could do what some chromebooks do and have the same PsysPLx calculation in coreboot as well.

Reply:
Generally BIOS should set the static maximum values for those power limits, so EC shouldn't set higher values than BIOS does. But if we end up in situation when we have to lower the power limits, then it should be called in some event handling function.

It is fine to set those values right after PCU reset if and only if the values are pretty much the same as the BIOS settings.

Reply:
> Hmm, the problem here is that setting the limits in coreboot is what enables the Psys functionality on the SoC side at all.

That's also right.

> I would prefer to only set them in the EC but that doesn't work.

That si why EC should set it asynchronically on some event function. Let's leave it like that, we can still fix things after some more testing will be done.